### PR TITLE
python310Packages.johnnycanencrypt: 0.6.0 -> 0.11.0

### PR DESCRIPTION
**Pull request by dotlambda**
ZHF: #199919 

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by risicle** [comment]
If you conditionally omit the `pcsclite` dependency this builds fine on macos 10.15 and you can remove the `broken` marker for that. Not everyone needs smartcard support.

**Comment by risicle** [comment]
`nixpkgs-review` happy, nixos x86_64 though

File: pkgs/applications/misc/tumpa/default.nix [hunk 1]
```diff
@@ -1,17 +1,12 @@
...
-, pyside2
-, johnnycanencrypt
-, pythonOlder
 , wrapQtAppsHook
 }:
 
-buildPythonPackage rec {
+python3.pkgs.buildPythonApplication rec {
   pname = "tumpa";
   version = "0.1.2";
```
Comment (0x4A6F) [inline pkgs/applications/misc/tumpa/default.nix:9]: This version AFAIK does not work with newer versions of `johnnycanencryt`, we can mark it broken and unbreak it, when https://github.com/kushaldas/tumpa/milestone/2 is completed.

Till then, `Generate -> Generate key` throws this error:
```
The key(s) not found in the keystore.
Traceback (most recent call last):
  File "/nix/store/k18rc9b6pg28xx2phdmr759jjisi248m-tumpa-0.1.2/lib/python3.10/site-packages/tumpasrc/__init__.py", line 480, in generate
    newk = self.ks.create_newkey(
AttributeError: 'KeyStore' object has no attribute 'create_newkey'. Did you mean: 'create_key'?
```

**Review by 0x4A6F — APPROVED** [review]
Changes LGTM.

**Review by SuperSandro2000 — APPROVED** [review]

**Comment by risicle** [comment]
`nixpkgs-review` happy, aarch64-linux.

**Pull request merged**